cmd/initialize_with_genesis.cpp

```diff
@@ -0,0 +1,258 @@
+    }
+    // We create the chaindata directory
+    fs::create_directories(out);
+
+    // We Initialize the database and open it
+    lmdb::DatabaseConfig db_config{out};
+    db_config.set_readonly(false);
+    std::shared_ptr<lmdb::Environment> env{lmdb::get_env(db_config)};
+    std::unique_ptr<lmdb::Transaction> txn{env->begin_rw_transaction()};
+
+    // Read genesis json file
+    nlohmann::json genesis_json;
+    nlohmann::json alloc_json = {};
+    if (chain_id <= 0) {
+        std::ifstream t(genesis.data());
+        std::string str((std::istreambuf_iterator<char>(t)),
+                        std::istreambuf_iterator<char>());
+        genesis_json = nlohmann::json::parse(str);
+        if (genesis_json.contains("alloc")) {
+            alloc_json = genesis_json["alloc"];
+        }
+    } else {
+        switch (chain_id) {
+            case 1:
+                genesis_json = kMainnetGenesis;
+                alloc_json = kMainnetGenesisAllocation;
+                break;
+            case 4:
+                // genesis_json = nlohmann::json::parse(kRinkebyGenesis);
+                break;
+            case 5:
+                // genesis_json = nlohmann::json::parse(kGoerliGenesis);
+                break;
+            default:
+                SILKWORM_LOG(LogLevel::Error) << "chain id: " << chain_id << " does not exist." << std::endl;
+                return -1;
+        }
+    }
+    if (!genesis_json.contains("difficulty") || !genesis_json.contains("nonce") ||
+        !genesis_json.contains("gasLimit") || !genesis_json.contains("timestamp") || 
+        !genesis_json.contains("extraData")) {
+        SILKWORM_LOG(LogLevel::Error) << "Incomplete Genesis File" << std::endl;
+        return -1;
+    }
+    // We create all tables
+    db::table::create_all(*txn);
+
+    auto block_number{Bytes(8, '\0')};
+    evmc::bytes32 root_hash;
+    if (alloc_json.size() > 0) {
+        // Filling account + constructing genesis root hash
+        std::map<evmc::bytes32, Bytes> account_rlp;
+        // Tables used
+        auto plainstate_table{txn->open(db::table::kPlainState)};
+        auto account_changeset_table{txn->open(db::table::kPlainAccountChangeSet)};
+        // Iterate over allocs
+        for (auto& [key, value] : alloc_json.items()) {
+            auto address_bytes{from_hex(key)};
+            if (address_bytes == std::nullopt) {
+                SILKWORM_LOG(LogLevel::Error) << "Cannot decode allocs from genesis" << std::endl;
+                return -1;
+            }
+            auto balance_str{value.get<std::string>()};
+            intx::uint256 balance;        
+            Account account;
+            if (is_hex(balance_str)) {
+                auto balance_bytes{from_hex(balance_str)};
+                auto [balance_decoded, err]{rlp::read_uint256(*balance_bytes, /*allow_leading_zeros=*/true)};
+                check_rlp_err(err);
+                balance = balance_decoded;
+            } else {
```

Comment:
Are you sure about this ?
`rlp::read_uint256(...)` swaps endianness and as far as I know the balance from alloc is an hex value **not**  rlp encoded.
The following `account.rlp(kEmptyRoot)` would reswap endianness back